pkg/dependencies/factory.go

```diff
@@ -951,7 +951,7 @@ func (f *Factory) clusterManagerOpts(timeoutOpts *ClusterManagerTimeoutOptions)
 }
 
 // WithClusterManager builds a cluster manager based on the cluster config and timeout options.
-func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster, timeoutOpts *ClusterManagerTimeoutOptions) *Factory {
+func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster, kubeconfig string, timeoutOpts *ClusterManagerTimeoutOptions) *Factory {
```

Comment:
instead of passing the kubeconfig here, I would suggest passing the `UnAuthKubeClient` to cluster manager and let it build the client with the kubeconfig in runtime

You can inject it as `ClientFactory`, which is an interface already defined in this package.